Lesson12/Lesson12_2.cpp

```diff
@@ -0,0 +1,103 @@
+// Lesson 12 2
+#include <iostream>
+
+enum class SortingDirection {ByRows, ByColumn};
+
+const int ROWS = 5;
+const int COLUMNS = 10;
+
+int partition(int array[], int low, int high)
+{
+    int pivot = array[high];
+
+    // idx of greater element
+    int pointerIdx = (low - 1);
+
+    for (int j = low; j < high; j++)
+    {
+        if (array[j] <= pivot)
+        {
+            // if element smaller than pivot is found
+            // swap it with the greater element pointed by i
+            pointerIdx++;
+
+            // swap element at i with element at j
+            std::swap(array[pointerIdx], array[j]);
+        }
+    }
+
+    std::swap(array[pointerIdx + 1], array[high]);
+
+    return (pointerIdx + 1);
+}
+
+void quickSort(int array[], int low, int high)
+{
+    if (low < high)
+    {
+        int partition_idx = partition(array, low, high);
+        quickSort(array, low, partition_idx - 1);
+        quickSort(array, partition_idx + 1, high);
+    }
+}
+
+void sort(int arr[ROWS][COLUMNS], SortingDirection sd)
+{
+    int arrByRow[ROWS];
+    int arrByCol[COLUMNS];
+    if(sd == SortingDirection::ByRows){
```

Comment:
you mixed up sort byRows and byColumns. sort by row is when your row index static and you change col index, so it is row: `a[0][0], a[0][1], a[0][2]`, and column is `a[0][0], a[1][0], a[2][0]`

Reply:
looks like fixed